x-pack/test/security_solution_cypress/cypress/e2e/detection_response/detection_engine/rule_creation/esql_rule.cy.ts

```diff
@@ -0,0 +1,302 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getEsqlRule } from '../../../../objects/rule';
+
+import {
+  RULES_MANAGEMENT_TABLE,
+  RULE_NAME,
+  INVESTIGATION_FIELDS_VALUE_ITEM,
+} from '../../../../screens/alerts_detection_rules';
+import {
+  RULE_NAME_HEADER,
+  RULE_TYPE_DETAILS,
+  RULE_NAME_OVERRIDE_DETAILS,
+  DEFINITION_DETAILS,
+  SUPPRESS_BY_DETAILS,
+  SUPPRESS_FOR_DETAILS,
+  SUPPRESS_MISSING_FIELD,
+} from '../../../../screens/rule_details';
+
+import { ESQL_QUERY_BAR } from '../../../../screens/create_new_rule';
+
+import { getDetails, goBackToRulesTable } from '../../../../tasks/rule_details';
+import { expectNumberOfRules } from '../../../../tasks/alerts_detection_rules';
+import { deleteAlertsAndRules } from '../../../../tasks/api_calls/common';
+import {
+  expandEsqlQueryBar,
+  fillAboutRuleAndContinue,
+  fillDefineEsqlRuleAndContinue,
+  fillScheduleRuleAndContinue,
+  selectEsqlRuleType,
+  getDefineContinueButton,
+  fillEsqlQueryBar,
+  fillAboutSpecificEsqlRuleAndContinue,
+  createRuleWithoutEnabling,
+  expandAdvancedSettings,
+  fillCustomInvestigationFields,
+  fillRuleName,
+  fillDescription,
+  getAboutContinueButton,
+  fillAlertSuppressionFields,
+  selectAlertSuppressionPerInterval,
+  setAlertSuppressionDuration,
+  selectDoNotSuppressForMissingFields,
+  continueFromDefineStep,
+  fillAboutRuleMinimumAndContinue,
+  skipScheduleRuleAction,
+  interceptEsqlQueryFieldsRequest,
+} from '../../../../tasks/create_new_rule';
+import { login } from '../../../../tasks/login';
+import { visit } from '../../../../tasks/navigation';
+
+import { CREATE_RULE_URL } from '../../../../urls/navigation';
+
+// https://github.com/cypress-io/cypress/issues/22113
+// issue is inside monaco editor, used in ES|QL query input
+// calling it after visiting page in each tests, seems fixes the issue
+// the only other alternative is patching ResizeObserver, which is something I would like to avoid
+const workaroundForResizeObserver = () =>
+  cy.on('uncaught:exception', (err) => {
+    if (err.message.includes('ResizeObserver loop limit exceeded')) {
+      return false;
+    }
+  });
+
+describe(
+  'Detection ES|QL rules, creation',
+  {
+    // skipped in MKI as it depends on feature flag alertSuppressionForEsqlRuleEnabled
+    // alertSuppressionForEsqlRuleEnabled feature flag is also enabled in a global config
+    tags: ['@ess', '@serverless', '@skipInServerlessMKI'],
+    env: {
+      kbnServerArgs: [
+        `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+          'alertSuppressionForEsqlRuleEnabled',
+        ])}`,
+      ],
+    },
+  },
+  () => {
+    const rule = getEsqlRule();
+    const expectedNumberOfRules = 1;
+
+    describe('creation', () => {
+      beforeEach(() => {
+        deleteAlertsAndRules();
+        login();
+      });
+
+      it('creates an ES|QL rule', function () {
+        visit(CREATE_RULE_URL);
```

Comment:
NIT: If you want, you can move from `visit` to `expandEsqlQueryBar` to the beforeEach hook of the `creation` describe.